### PR TITLE
Re-enable IP-in-IP ST testing

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

File: calico_node/tests/st/bgp/test_ipip.py [hunk 1]
```diff
@@ -402,21 +400,53 @@ def _test_gce_int(self, with_ipip, backend, host1, host2, rrc):
...
+                    # running 'ip monitor' for 10s and checking that it does
+                    # not indicate any changes other than those associated with
+                    # an IP address being added to the tunl0 device.
+                    ip_changes = host.execute("timeout -t 10 ip -t monitor 2>&1 || true")
+                    lines = ip_changes.split("\n")
+                    assert len(lines) >= 1, "No output from ip monitor"
+                    expected_lines_following = 0
+                    for line in lines:
+                        expected = False
+                        if expected_lines_following > 0:
```
Comment (tmjd) [inline calico_node/tests/st/bgp/test_ipip.py]: The `expected_lines_following` could be avoided by just looping on an index variable and then any case that expects additional trailing lines could just advance the index the correct number of additional lines.  I think that might make the loop more readable, WDYT?  Though I guess looping with an index like that is less 'pythonic' than the current loop.

File: calico_node/tests/st/utils/docker_host.py [hunk 2]
```diff
@@ -319,12 +321,16 @@ def start_calico_node(self, options="", with_ipv4pool_cidr_env_var=True):
...
+            if FELIX_LOGLEVEL != "":
+                felix_logsetting = " -e FELIX_LOGSEVERITYSCREEN=" + FELIX_LOGLEVEL
+
             # Construct the calicoctl command that we want, including the
             # CALICO_IPV4POOL_CIDR setting.
             modified_cmd = (
                 prefix +
                 (" -e CALICO_IPV4POOL_CIDR=%s " % DEFAULT_IPV4_POOL_CIDR) +
-                " -e DISABLE_NODE_IP_CHECK=true -e FELIX_IPINIPENABLED=true " +
+                " -e DISABLE_NODE_IP_CHECK=true" + felix_logsetting + " -e FELIX_IPINIPENABLED=true " +
```
Comment (tmjd) [inline calico_node/tests/st/utils/docker_host.py]: I think you should add this on its own line because I expect we will remove the DISABLE_NODE_IP_CHECK and FELIX_IPIPENABLED flags at some point (soon) and I'd rather the two changes not be mixed.

File: calico_node/tests/st/utils/docker_host.py [hunk 3]
```diff
@@ -670,8 +676,10 @@ def _exec_calicoctl(self, action, data):
...
         # Run a set of commands to trace ip routes, iptables and ipsets.
         self.execute("ip route", raise_exception_on_failure=False)
-        self.execute("iptables-save", raise_exception_on_failure=False)
-        self.execute("ip6tables-save", raise_exception_on_failure=False)
+        self.execute("iptables-save -c", raise_exception_on_failure=False)
+        self.execute("ip6tables-save -c", raise_exception_on_failure=False)
         self.execute("ipset save", raise_exception_on_failure=False)
         self.execute("ps", raise_exception_on_failure=False)
-        self.execute("cat /etc/bird/bird.conf", raise_exception_on_failure=False)
+        self.execute("docker logs calico-node", raise_exception_on_failure=False)
```
Comment (tmjd) [inline calico_node/tests/st/utils/docker_host.py]: Why did you remove dumping the bird.conf? I think it should remain unless you know that it isn't useful anymore.

File: calico_node/tests/st/bgp/test_ipip.py (hunk 1 above)
Comment (neiljerram) [reply calico_node/tests/st/bgp/test_ipip.py]: I think I prefer to keep as is.  Please see what you think now that I've removed `expected` - I think it looks reasonably neat.

File: calico_node/tests/st/utils/docker_host.py (hunk 3 above)
Comment (neiljerram) [reply calico_node/tests/st/utils/docker_host.py]: The line as was did not work for two reasons:

1. The BIRD config is inside the calico-node container, not directly on the host.

2. The BIRD config is not at that path, even within the container.

I think a correct line would be something like
```
self.execute("docker exec calico-node cat /etc/calico/confd/bird.cfg", raise_exception_on_failure=False)
```
but I don't currently have reason to need that.
￼

**Comment by tmjd** [comment]
Don't forget to squash before merging.
